### PR TITLE
Add FeatureSever to STAC

### PR DESCRIPTION
**Pull request by smohiudd**
- Various changes to MapServer and ImageServer to make valid stac collection
- Checks for `GetCapabilities` in MapServer and ImageServer. If 400 error then assume there is no WMSServer and cannot generate stac collection
- Add FeatureSever to STAC
- Adds `web-map-links ` and `datacube` extensions
- Add tests
- formatting and sort imports

### CODE REVIEW

**Comment by amarouane-ABDELHAK** [comment]
Looks good, thank you for working on this! Could we add a Jupyter Notebook example for FeatureServer_to_STAC, similar to [these examples](https://github.com/NASA-IMPACT/pyarc2stac/tree/main/examples)?

**Comment by smohiudd** [comment]
@amarouane-ABDELHAK sure I'll add it to the notebooks

**Review by amarouane-ABDELHAK — APPROVED** [review]

File: pyarc2stac/utils.py [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
 
+def get_xml(url) -> ET:
+    r = requests.get(url)
+    r.raise_for_status()
+    return ET.fromstring(r.content)
+
+
 def convert_to_datetime(times_extent):
     return [
         datetime.utcfromtimestamp(time_extent / 1000.0) for time_extent in times_extent
```
Comment (anayeaye) [inline pyarc2stac/utils.py]:
```suggestion
        datetime.fromtimestamp((time_extent/1000), UTC) for time_extent in times_extent
```

File: pyarc2stac/utils.py [hunk 2]
```diff
@@ -1,14 +1,22 @@
+import xml.etree.ElementTree as ET
 from datetime import datetime
```
Comment (anayeaye) [inline pyarc2stac/utils.py]:
```suggestion
from datetime import datetime, UTC
```

**Review by anayeaye — APPROVED** [review]
Approve, but for python3.12 these changes are needed in pyarc2stac/utils.py to run the tests locally
```
-from datetime import datetime
+from datetime import datetime, UTC
 
 def convert_to_datetime(times_extent):
     return [
-        datetime.utcfromtimestamp(time_extent / 1000.0) for time_extent in times_extent
+        datetime.fromtimestamp((time_extent/1000), UTC) for time_extent in times_extent
     ]
```

**Pull request merged**